src/main/flight/pid.c

```diff
@@ -664,15 +669,24 @@ void pidInitConfig(const pidProfile_t *pidProfile)
         thrustLinearizationB = (1.0f - thrustLinearization) / (2.0f * thrustLinearization);
     }
 #endif    
-#if defined(USE_D_CUT)
-    dtermCutPercent = pidProfile->dterm_cut_percent / 100.0f;
-    dtermCutPercentInv = 1.0f - dtermCutPercent;
-    dtermCutRangeHz = pidProfile->dterm_cut_range_hz;
-    dtermCutLowpassHz = pidProfile->dterm_cut_lowpass_hz;
-    dtermCutGain = pidProfile->dterm_cut_gain * dtermCutPercent * D_CUT_GAIN_FACTOR / dtermCutLowpassHz;
+#if defined(USE_D_MIN)
+    if ((pidProfile->d_min_roll > 0) && (pidProfile->d_min_roll < pidProfile->pid[FD_ROLL].D)){
+        dMinPercentRoll = (pidProfile->d_min_roll) / (float)(pidProfile->pid[FD_ROLL].D);
+    } else {
+    dMinPercentRoll = 0;
+    }
+    if ((pidProfile->d_min_pitch > 0) && (pidProfile->d_min_pitch < pidProfile->pid[FD_PITCH].D)){
+        dMinPercentPitch = (pidProfile->d_min_pitch / (float)pidProfile->pid[FD_PITCH].D);
+    } else {
+    dMinPercentPitch = 0;
+    }
+    dMinGyroGain = pidProfile->d_min_gain * D_MIN_GAIN_FACTOR / D_MIN_LOWPASS_HZ;
+    dMinSetpointGain = pidProfile->d_min_gain * D_MIN_SETPOINT_GAIN_FACTOR * pidProfile->d_min_advance * pidFrequency / (100 * D_MIN_LOWPASS_HZ);
     // lowpass included inversely in gain since stronger lowpass decreases peak effect
+    dMinGyroFactor = 1.0f;
+    dMinSetpointFactor = 1.0f;
 #endif
-    dtermCutFactor = 1.0f;
+    dMinFactor = 1.0f;
```

Comment:
I don't like that 'dMinFactor' leaks into the rest of the code when `USE_D_MIN` undef.

Reply:
I did it like that so there was no need for a conditional build of the D calculation itself...
Currently it is 
```
pidData[axis].D = pidCoefficient[axis].Kd * delta * tpaFactor * dMinFactor;
```
I could put dMinFactor into ifdef if the main D calculation also was, eg:
```
#if defined(USE_D_MIN)
pidData[axis].D = pidCoefficient[axis].Kd * delta * tpaFactor * dMinFactor;
#else
pidData[axis].D = pidCoefficient[axis].Kd * delta * tpaFactor
#endif
```
Should I do this?

Reply:
Yes, that seems solid to me. Drops the multiplication when it's undef, too.

Reply:
OK, uploaded and fixed, thanks!